src/IO/ADIOS/ADIOS2Auxiliary.cpp

```diff
@@ -0,0 +1,172 @@
+#include "openPMD/config.hpp"
+#if openPMD_HAVE_ADIOS2
+#include "openPMD/IO/ADIOS/ADIOS2Auxiliary.hpp"
+#include "openPMD/Datatype.hpp"
+#include <iostream>
+
+namespace openPMD
+{
+namespace detail
+{
+    template< typename T >
+    std::string
+    ToDatatypeHelper< T >::type()
+    {
+        return adios2::GetType< T >();
+    }
+
+    template< typename T >
+    std::string
+    ToDatatypeHelper< std::vector< T > >::type()
+    {
+        return
+
+            adios2::GetType< T >();
+    }
+
+    template< typename T, size_t n >
+    std::string
+    ToDatatypeHelper< std::array< T, n > >::type()
+    {
+        return
+
+            adios2::GetType< T >();
+    }
+
+    std::string
+    ToDatatypeHelper< bool >::type()
+    {
+        return ToDatatypeHelper< bool_representation >::type();
+    }
+
+    template< typename T >
+    std::string
+    ToDatatype::operator()()
+    {
+        return ToDatatypeHelper< T >::type();
+    }
+
+    template< int n >
+    std::string
+    ToDatatype::operator()()
+    {
+        return "";
+    }
+
+    Datatype
+    fromADIOS2Type( std::string const & dt )
+    {
+        static std::map< std::string, Datatype > map{
+            { "string", Datatype::STRING },
+            { "char", Datatype::CHAR },
+            { "signed char", Datatype::CHAR },
+            { "unsigned char", Datatype::UCHAR },
+            { "short", Datatype::SHORT },
+            { "unsigned short", Datatype::USHORT },
+            { "int", Datatype::INT },
+            { "unsigned int", Datatype::UINT },
+            { "long int", Datatype::LONG },
+            { "unsigned long int", Datatype::ULONG },
+            { "long long int", Datatype::LONGLONG },
+            { "unsigned long long int", Datatype::ULONGLONG },
+            { "float", Datatype::FLOAT },
+            { "double", Datatype::DOUBLE },
+            { "long double", Datatype::LONG_DOUBLE },
+            { "uint8_t", Datatype::UCHAR },
```

Comment:
no (u)char type in adios2?

Reply:
Not any longer, see [here](https://github.com/ornladios/ADIOS2/pull/1452).

Reply:
hm...